pkg/cmd/render/render.go

```diff
@@ -162,6 +169,22 @@ func (r *renderOpts) Run() error {
 			return fmt.Errorf("unable to parse restricted CIDRs from config %q: %v", r.clusterConfigFile, err)
 		}
 	}
+	if len(r.clusterAuthFile) > 0 {
+		clusterAuthFileData, err := ioutil.ReadFile(r.clusterAuthFile)
+		if err != nil && !os.IsNotExist(err) {
+			return fmt.Errorf("failed to load authentication config: %v", err)
+		}
+		if len(clusterAuthFileData) > 0 {
+			if err := discoverServiceAccountIssuer(clusterAuthFileData, &renderConfig); err != nil {
+				return fmt.Errorf("unable to parse service-account issuers from config %q: %v", r.clusterAuthFile, err)
+			}
+		}
+	}
+	if _, err := os.Stat(filepath.Join(r.generic.AssetInputDir, "bound-service-account-signing-key.key")); err == nil {
+		if _, err := os.Stat(filepath.Join(r.generic.AssetInputDir, "bound-service-account-signing-key.pub")); err == nil {
+			renderConfig.UserProvidedBoundSASigningKey = true
```

Comment:
Would it make sense to validate the keypair here to ensure that the user provided keypair is valid? Not doing so risks an apiserver that can't start. Given that with RSA it's possible to find the public key given a private key, would it make sense to only accept the private key?

Reply:
the installer ensure that the key-pair are valid and only valid pairs end up on the bootstrap host, see https://github.com/openshift/installer/pull/4373#issuecomment-728219165

so i think we can assume valid inputs here. kap-o render is mostly designed for bootstrap host and i think we should be good making this assumption.